cmd/platform/sampledata.go

```diff
@@ -0,0 +1,628 @@
+	if err != nil {
+		return errors.New("Invalid profile-images parameter")
+	}
+	profileImages := []string{}
+	if profileImagesPath != "" {
+		profileImagesStat, err := os.Stat(profileImagesPath)
+		if os.IsNotExist(err) {
+			return errors.New("Profile images folder doesn't exists.")
+		}
+		if !profileImagesStat.IsDir() {
+			return errors.New("profile-images parameters must be a folder path.")
+		}
+		profileImagesFiles, err := ioutil.ReadDir(profileImagesPath)
+		if err != nil {
+			return errors.New("Invalid profile-images parameter")
+		}
+		for _, profileImage := range profileImagesFiles {
+			profileImages = append(profileImages, path.Join(profileImagesPath, profileImage.Name()))
+		}
+		sort.Strings(profileImages)
+	}
+
+	if workers < 1 {
+		return errors.New("You must have at least one worker.")
+	}
+	if teamMemberships > teams {
+		return errors.New("You can't have more team memberships than teams.")
+	}
+	if channelMemberships > channelsPerTeam {
+		return errors.New("You can't have more channel memberships than channels per team.")
+	}
+
+	var bulkFile *os.File
+	switch bulk {
+	case "":
+		bulkFile, err = ioutil.TempFile("/tmp", "mattermost-sample-data-")
+		defer os.Remove(bulkFile.Name())
+		if err != nil {
+			return errors.New("Unable to open temporary file.")
+		}
+	case "-":
+		bulkFile = os.Stdout
+	default:
+		bulkFile, err = os.OpenFile(bulk, os.O_RDWR|os.O_CREATE, 0755)
+		if err != nil {
+			return errors.New("Unable to write into the \"" + bulk + "\" file.")
+		}
+	}
+
+	encoder := json.NewEncoder(bulkFile)
+	version := 1
+	encoder.Encode(app.LineImportData{Type: "version", Version: &version})
+
+	fake.Seed(seed)
+	rand.Seed(seed)
+
+	teamsAndChannels := make(map[string][]string)
+	for i := 0; i < teams; i++ {
+		teamLine := createTeam(i)
+		teamsAndChannels[*teamLine.Team.Name] = []string{}
+		encoder.Encode(teamLine)
+	}
+
+	teamsList := []string{}
+	for teamName := range teamsAndChannels {
+		teamsList = append(teamsList, teamName)
+	}
+	sort.Strings(teamsList)
+
+	for _, teamName := range teamsList {
+		for i := 0; i < channelsPerTeam; i++ {
+			channelLine := createChannel(i, teamName)
+			teamsAndChannels[teamName] = append(teamsAndChannels[teamName], *channelLine.Channel.Name)
+			encoder.Encode(channelLine)
+		}
+	}
+
+	allUsers := []string{}
+	for i := 0; i < users; i++ {
+		userLine := createUser(i, teamMemberships, channelMemberships, teamsAndChannels, profileImages)
+		encoder.Encode(userLine)
+		allUsers = append(allUsers, *userLine.User.Username)
+	}
+
+	for team, channels := range teamsAndChannels {
+		for _, channel := range channels {
+			for i := 0; i < postsPerChannel; i++ {
+				postLine := createPost(team, channel, allUsers)
+				encoder.Encode(postLine)
+			}
+		}
+	}
+
+	for i := 0; i < directChannels; i++ {
+		user1 := allUsers[rand.Intn(len(allUsers))]
+		user2 := allUsers[rand.Intn(len(allUsers))]
+		channelLine := createDirectChannel([]string{user1, user2})
```

Comment:
Looks like theoretically the same channel could be generated twice. Does this actually matter, or does the last one generated just take precedence when importing?

Reply:
Exactly, this currently is not perfect, because can generate "less channels than requested", but the final generated bulk works, because the first one create the channel and the second one update it.

Reply:
That's fine by me.